tests/integration_tests/test_catboost.py

```diff
@@ -123,6 +123,17 @@ def objective(trial: optuna.trial.Trial) -> float:
 
     # Unknown validation name or metric.
     study = optuna.create_study(pruner=DeterministicPruner(False))
-    # Catboost terminates with a SystemError.
-    with pytest.raises(SystemError):
-        study.optimize(objective, n_trials=1)
+    # Catboost terminates with a SystemError when python>=3.9 or pytest>=7.2.0, otherwise terminates with RecursionError.
+    # This is mainly because the _Py_CheckRecursionLimit variable used in from limited C API was removed after python 3.9.
+    # That affect when calling function _CallbackAfterIteration in Catboost, which is written in Cython.
```

Comment:
```suggestion
    # Catboost terminates with a SystemError when python>=3.9 or pytest>=7.2.0,
    # otherwise terminates with RecursionError.
    # This is mainly because the _Py_CheckRecursionLimit variable used in from
    # limited C API was removed after python 3.9.
    # That affect when calling function _CallbackAfterIteration in Catboost,
    # which is written in Cython.
```
It seems that the CI is failing due to the long comment.